pkg/issuer/acme/setup.go

```diff
@@ -102,20 +104,19 @@ func (a *Acme) Setup(ctx context.Context) error {
 		s := messageAccountVerificationFailed + err.Error()
 		apiutil.SetIssuerCondition(a.issuer, v1alpha2.IssuerConditionReady, cmmeta.ConditionFalse, errorAccountVerificationFailed, s)
 		return fmt.Errorf(s)
-
 	}
-
-	acme.ClearClientCache()
-
-	cl, err := acme.ClientWithKey(a.issuer, pk)
-	if err != nil {
-		s := messageAccountVerificationFailed + err.Error()
-		log.Error(err, "failed to verify acme account")
-		a.Recorder.Event(a.issuer, v1.EventTypeWarning, errorAccountVerificationFailed, s)
-		apiutil.SetIssuerCondition(a.issuer, v1alpha2.IssuerConditionReady, cmmeta.ConditionFalse, errorAccountVerificationFailed, s)
-		return err
+	rsaPk, ok := pk.(*rsa.PrivateKey)
+	if !ok {
+		apiutil.SetIssuerCondition(a.issuer, v1alpha2.IssuerConditionReady, cmmeta.ConditionFalse, errorAccountVerificationFailed, fmt.Sprintf("ACME private key in %q is not of type RSA", a.issuer.GetSpec().ACME.PrivateKey.Name))
+		return nil
 	}
 
+	// TODO: don't always clear the client cache.
+	//  In future we should intelligently manage items in the account cache
+	//  and remove them when the corresponding issuer is updated/deleted.
+	a.accountRegistry.RemoveClient(string(a.issuer.GetUID()))
```

Comment:
Hm, I think this is actually a change in behaviour here that I didn't intend with this PR. We need to call an equivalent of `ClearClientCache` else we will never remove clients once an issuer has been *deleted*. Perhaps instead of doing that here though, I can do it in my follow up PR that adds a dedicated controller to manage the cache :)

Reply:
As we spoke about, I also would like to see similar behavior for the cleaning up Certificate metrics once they are deleted